File: csharp/main/LiteralEnumerationValue.cs
using System;

namespace Deltix.Luminary
{
	public class LiteralEnumerationValue : Literal, ILiteral<EnumerationMemberDef>
	{
		public LiteralEnumerationValue(EnumerationMemberDef value)
			: base(LiteralKind.EnumerationValue)
		{
			Value = value;
		}

		public EnumerationMemberDef Value { get; }

		public override String ToString()
		{
			return $"{Value.Owner.Name}.{Value.Name}";
		}
	}
}

File: java/src/main/java/deltix/luminary/LiteralEnumerationValue.java
package deltix.luminary;

public class LiteralEnumerationValue extends Literal {
    private final EnumerationMemberDef value;

    public LiteralEnumerationValue(EnumerationMemberDef value) {
        super(LiteralKind.ENUMERATION_VALUE);
        this.value = value;
    }

    public EnumerationMemberDef getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value.getOwner().getName() + "." + value.getName();
    }
}
